File: project/sumin-onboarding/survey-service/domain/src/main/java/com/innercircle/survey/domain/entity/SurveyResponse.java
package com.innercircle.survey.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "survey_response")
@Getter
@NoArgsConstructor
public class SurveyResponse {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id", nullable = false)
    private Survey survey;

    //제출된 응답들(JSON)
    @Lob
    @Column(nullable = false)
    private String answersJson;

    //질문 스냅샷(JSON)
    @Lob
    @Column(nullable = false)
    private String snapshotJson;

    @Column(nullable = false, updatable = false)
    private LocalDateTime submittedAt;

    @PrePersist
    public void prePersist() {
        this.submittedAt = LocalDateTime.now();
    }

    public SurveyResponse(Survey survey, String answersJson, String snapshotJson) {
        this.survey = survey;
        this.answersJson = answersJson;
        this.snapshotJson = snapshotJson;
    }
}

File: project/kyungrok-onboarding/src/main/kotlin/kr/innercircle/onboarding/survey/domain/SurveyResponse.kt
package kr.innercircle.onboarding.survey.domain

import jakarta.persistence.*
import kr.innercircle.onboarding.survey.dto.request.CreateSurveyResponseRequest

/**
 * packageName : kr.innercircle.onboarding.survey.domain
 * fileName    : SurveyResponse
 * author      : ckr
 * date        : 25. 6. 12.
 * description :
 */

@Table(name = "survey_response")
@Entity
class SurveyResponse(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id", nullable = false)
    var survey: Survey,

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false , length = 100)
    var respondent: String
): BaseTimeEntity() {
    companion object {
        fun from(
            survey: Survey,
            createSurveyResponseRequest: CreateSurveyResponseRequest
        ): SurveyResponse {
            return SurveyResponse(
                survey = survey,
                respondent = createSurveyResponseRequest.respondent
            )
        }
    }
}